cli/main.rs

```diff
@@ -1477,8 +1494,7 @@ pub fn main() {
 
   logger::init(flags.log_level);
 
-  unwrap_or_exit(run_basic(get_subcommand(flags)));
+  let exit_code = unwrap_or_exit(run_basic(get_subcommand(flags)));
 
-  let code = deno_runtime::EXIT_CODE.load(Relaxed);
-  std::process::exit(code);
+  std::process::exit(exit_code);
```

Comment:
I think we should only call `exit` if `exit_code` is not equal to 0. Otherwise drop handlers will not be called which might lead to zombie subprocesses.

Reply:
What drop handlers are you worried about? The real work is scoped to run_basic(), right?

Reply:
All drop handlers - IIRC if you call `std::process:exit` the process immediately terminates and no drop handlers are run; when we run a subprocess using `Deno.run()` we got `close_on_drop` set to true, and when you are existing using `exit()` these drop handlers will not run and subprocesses will not close becoming zombies.

Reply:
I'm still confused. Are you talking about lexically scoped drop handlers? main() doesn't have any that matter, does it?

(I'm also confused by your `close_on_drop` comment - where is that a thing inside the source tree?)

Reply:
Please scratch that, I should have commented before the first coffee :) of course drop handlers run because they are scoped to `run_basic()`. And in case of `close_on_drop` I actually meant [`kill_on_drop`](https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.kill_on_drop)